### ISSUE TITLE
Book UI

### ISSUE THREAD

**Issue by m00racle** [m00racle/refract_book#2]
This will focus more on the transactions on the specific books

- [x] design all possible transaction steps
- [x] how to accommodate debit credit side
- [x] how to accommodate documents related to transaction
- [x] how to choose component to show according to state induced by user inputs
- [x] how to create tabs in order to isolate transaction and reporting activities
- [x] #4
- [x] CANCELED: ~~how to create reporting of specific book~~ <-This requires separate study

**Comment by m00racle** [comment]
I think for starting point this [Codelab: Expense Tracker](https://github.com/m00racle/expense-tracker) will be a good reference

**Comment by m00racle** [comment]
I still considering the use of drawer at the moment. But if this is too much for a MVP I will just pass this out and delete (for now) the drawer. 

I still consider to just put multiple link  on the navigation bar and perhaps nested for documents so that the user can choose what document he/she want to input.

**Comment by m00racle** [comment]
## LOCALIZATION
source: [Material UI localization](https://mui.com/material-ui/guides/localization/)

I need to understand what is localization and how to integrate the localization to our app.

Material-UI provides built-in support for localization, allowing you to create applications that can be adapted to different languages, regions, and cultural conventions. Localization in Material-UI involves translating text, formatting dates, numbers, and currencies, and handling other locale-specific behaviors.

To implement localization in Material-UI, you can follow these steps:

1. Install the required dependencies:
   - `@formatjs/intl`: Provides internationalization support.
   - `@formatjs/intl-pluralrules`: Polyfills for `Intl.PluralRules` API.
   - `@formatjs/intl-relativetimeformat`: Polyfills for `Intl.RelativeTimeFormat` API.

2. Set up a localization context in your application. You can create a `LocalizationContext` using the `createIntl` function from `@formatjs/intl` package. This context should be provided to the component tree where localization is needed.

3. Define language-specific message strings for your application. These message strings can be stored in separate JSON files for each language or loaded dynamically from an API.

4. Use the `FormattedMessage` component from `react-intl` to render localized text. The `FormattedMessage` component takes the message string as a `id` prop and handles the translation based on the current locale.

5. Format dates, numbers, and currencies using the `intl` object from the localization context. You can access the `intl` object using the `useIntl` hook from `react-intl`. The `intl` object provides methods for formatting values based on the current locale, such as `formatDate`, `formatNumber`, and `formatRelativeTime`.

Here's a simplified example of implementing localization in Material-UI:

```jsx
import React from 'react';
import { createIntl, createIntlCache, RawIntlProvider, useIntl } from 'react-intl';

// Define language-specific message strings
const messages = {
  en: {
    greeting: 'Hello, {name}!',
  },
  fr: {
    greeting: 'Bonjour, {name} !',
  },
};

// Create the localization context
const cache = createIntlCache();
const intl = createIntl(
  {
    locale: 'en', // Set the initial locale
    messages,
  },
  cache
);

// Component that uses localized text
function Greeting({ name }) {
  const { formatMessage } = useIntl();
  const greeting = formatMessage({ id: 'greeting' }, { name });

  return <p>{greeting}</p>;
}

// App component
function App() {
  return (
    <RawIntlProvider value={intl}>
      <Greeting name="John" />
    </RawIntlProvider>
  );
}

export default App;
```

In this example, the `Greeting` component renders a localized greeting based on the current locale. The `useIntl` hook is used to access the `formatMessage` function, which translates the message string based on the provided locale.

By setting the `locale` prop of the `createIntl` function, you can dynamically change the locale and trigger re-rendering with the updated translations.

This is just a basic overview of how to implement localization in Material-UI. You can explore more advanced features and techniques, such as pluralization, date formatting, and language switchers, by referring to the official documentation and examples provided by Material-UI and the `react-intl` library.

**Comment by m00racle** [comment]
## Add tabs

Here is the simple tabs design from Maerial UI:

```jsx
import * as React from 'react';
import PropTypes from 'prop-types';
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';

function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box sx={{ p: 3 }}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
}

TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.number.isRequired,
  value: PropTypes.number.isRequired,
};

function a11yProps(index) {
  return {
    id: `simple-tab-${index}`,
    'aria-controls': `simple-tabpanel-${index}`,
  };
}

export default function BasicTabs() {
  const [value, setValue] = React.useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  return (
    <Box sx={{ width: '100%' }}>
      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
        <Tabs value={value} onChange={handleChange} aria-label="basic tabs example">
          <Tab label="Item One" {...a11yProps(0)} />
          <Tab label="Item Two" {...a11yProps(1)} />
          <Tab label="Item Three" {...a11yProps(2)} />
        </Tabs>
      </Box>
      <TabPanel value={value} index={0}>
        Item One
      </TabPanel>
      <TabPanel value={value} index={1}>
        Item Two
      </TabPanel>
      <TabPanel value={value} index={2}>
        Item Three
      </TabPanel>
    </Box>
  );
}
```

**Comment by m00racle** [comment]
### Explain on the code above:
`propTypes` is a property used in React to define the expected types of the component's props. It is not a standard feature of JavaScript itself, but rather a feature provided by React.

In the code you provided, `propTypes` is used to define the expected types of the props for the `TabPanel` component. It helps with type-checking and provides a way to communicate the expected prop types to other developers who may be using or extending the component.

Here's a breakdown of the code and how `propTypes` is used:

```jsx
TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.number.isRequired,
  value: PropTypes.number.isRequired,
};
```

In this code, `TabPanel.propTypes` is assigned an object that specifies the prop types for the `TabPanel` component. Each key-value pair in the object represents a prop, where the key is the prop name and the value is the corresponding prop type.

In this case, the `children` prop is specified as `PropTypes.node`, which means it can accept any renderable content. The `index` and `value` props are specified as `PropTypes.number.isRequired`, indicating that they must be of type `number` and they are required (cannot be omitted).

The `PropTypes` object used here is part of the React library. It provides a set of utilities for defining prop types and performing prop type validations.

Using `propTypes` is not mandatory, but it is considered a good practice for documenting and enforcing the expected prop types in React components. It helps catch potential bugs and provides better development experience and maintainability.

**Comment by m00racle** [comment]
Does onChange function that pass newValue only exclusive to Tabs React component or other React components as well?

The `onChange` function and the arguments it receives can vary depending on the component or element you're working with. While the `Tabs` component in Material-UI provides the `newValue` argument, it's not a universal standard for all input elements.

For `<input>` elements, such as text inputs or checkboxes, the `onChange` event typically provides an `event` object as the argument. You can access the new value using `event.target.value`. Here's an example for an `<input>` element:

```jsx
const handleChange = (event) => {
  const newValue = event.target.value;
  // Use the `newValue` here
};

<input type="text" value={value} onChange={handleChange} />
```

For `<select>` elements, the `onChange` event also provides an `event` object as the argument. You can access the selected value using `event.target.value`. Here's an example for a `<select>` element:

```jsx
const handleChange = (event) => {
  const newValue = event.target.value;
  // Use the `newValue` here
};

<select value={value} onChange={handleChange}>
  {/* Options */}
</select>
```

In summary, the `onChange` function and the arguments it receives depend on the specific component or element you're working with. While the `Tabs` component in Material-UI provides the `newValue` argument, other input elements like `<input>` and `<select>` typically provide an `event` object from which you can extract the new value.

**Comment by m00racle** [comment]
### WARNING: React Component Function Name
I got bug that the contents are displayed on all panels not to specific panel related to the content; then there is a WARNING: use Pascal case for React component naming

The warning "use Pascal case for React components" means that you should use capital letters for the names of your React component functions or classes.

In JavaScript and JSX, it's a convention to use Pascal case (also known as upper camel case) for naming React components. This means that the first letter of each word in the component name should be capitalized.

For example, instead of naming your component `reportPanel`, you should name it `ReportPanel`. Similarly, instead of naming your component `allyProps`, you should name it `AllyProps`.

Here's an updated version of your code with the component names corrected:

```jsx
// ...

function ReportPanel(props) {
    // ...
}

// Specify the ReportPanel props
ReportPanel.propTypes = {
    children: PropTypes.node,
    index: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
};

// Ally props for aria
function AllyProps(index) {
    // ...
}

export default function DocumentPage() {
    // ...
}

// ...
```

By using Pascal case for your component names, you follow the standard convention and avoid the warning about using Pascal case for React components.
